library/manageiq_policy.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python
+
+import os
+from ansible.module_utils.basic import *
+from miqclient.api import API as MiqApi
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_policy
+short_description: assign and unassign policies and policy profiles on resources in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  entity:
+    description:
+      - the entity referred to
+    required: true
+    choices: ['policy', 'policy profile']
+  entity_name:
+    description:
+      - the entity name in manageiq
+    required: true
+    default: null
+  resource:
+    description:
+      - the relevant resource type in manageiq
+    required: true
+    choices: ['provider', 'host', 'vm', 'node', 'pod', 'replicator', 'container images']
+    default: null
+  resource_name:
+    description:
+      - the relevant resource name in manageiq
+    required: true
+    default: null
+  action:
+    description:
+      - the action used for managing the entity on the resource
+      - On assign, it will assign the entity on the resource, if not already assigned
+      - On absent, it will unassign the entity from the resource
+    required: false
+    choices: ['assign', 'unassign']
+    default: 'assign'
+'''
+
+EXAMPLES = '''
+# Assign policy to a provider in ManageIQ
+  manageiq_policy:
+    entity: 'policy'
+    entity_name: 'example policy'
+    resource_name: 'OpenShift01'
+    resource: 'provider'
+    action: 'assign'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+'''
+
+
+class ManageIQ(object):
+    """ ManageIQ object to execute policy related actions in manageiq
+
+    url      - manageiq environment url
+    user     - the username in manageiq
+    password - the user password in manageiq
+    """
+
+    manageiq_entities = {
+        'policy': 'policies', 'policy profile': 'policy_profiles',
+        'provider': 'providers', 'host': 'hosts', 'vm': 'vms',
+        'node': 'container_nodes', 'pod': 'container_pods',
```

Comment:
> By what criterion does only "container image" have "container " prefix?

Since assigning policies is currently supported only for *container* images I didnt want users to confuse them with vm images for example.

> In UI I think Container Image and Container Node have "Container " prefix. (Though that might change in future.)

I'll also add the prefix for nodes. @simon3z any thoughts?  

> Does container_pods actually work with the API? Shouldn't it be container_groups?

it won't work, I'll make the change, thanks!